src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -377,53 +359,118 @@ def work_for_vts_rsc(vts, ctx):
           ' (',
           ctx.target.address.spec,
           ').')
+        # things to do
+        # collect jar dependencies outlines
+        # collect non-java dependencies
+        # metacp remaining java dependencies using jar and non-java deps as deps
+        # rsc sources,
+        #   providing metacp'd
+        #    - jar deps
+        #    - non-java deps
+        #    - metacp'd java deps
+
+
+
+        strict_dependencies_for_target = list(DependencyContext.global_instance().dependencies_respecting_strict_deps(target))
+
+        jar_deps = [t for t in strict_dependencies_for_target if isinstance(t, JarLibrary)]
+
+        def is_java_compile_target(t):
+          return isinstance(t, JavaLibrary) or t.has_sources('.java')
+        java_deps = [t for t in strict_dependencies_for_target
+                     if is_java_compile_target(t)]
+        non_java_deps = [t for t in strict_dependencies_for_target
+                         if not (is_java_compile_target(t)) and not isinstance(t, JarLibrary)]
+
+        metacped_jar_classpath_abs = _paths_from_classpath(
+          self._metacp_jars_classpath_product.get_for_targets(jar_deps)
+        )
+        metacped_jar_classpath_abs.extend(self._jvm_lib_metacp_classpath)
+        metacped_jar_classpath_rel = fast_relpath_collection(metacped_jar_classpath_abs)
+
+        jar_rsc_classpath_paths = _paths_from_classpath(
+          self.context.products.get_data('rsc_classpath').get_for_targets(jar_deps),
+          collection_type=set)
+
+        non_java_paths = _paths_from_classpath(
+          self.context.products.get_data('rsc_classpath').get_for_targets(non_java_deps),
+          collection_type=set)
+        non_java_rel = fast_relpath_collection(non_java_paths)
 
-        tgt, = vts.targets
+        java_paths = _paths_from_classpath(
+          self.context.products.get_data('rsc_classpath').get_for_targets(java_deps),
+          collection_type=set)
+        rel_java_paths = fast_relpath_collection(java_paths)
+
+        ctx.ensure_output_dirs_exist()
+
+        distribution = self._get_jvm_distribution()
         with Timer() as timer:
-          if cp_entries:
-            # Step 1: Convert classpath to SemanticDB
-            # ---------------------------------------
-            scalac_classpath_path_entries_abs = self.tool_classpath('workaround-metacp-dependency-classpath')
-            scalac_classpath_path_entries = fast_relpath_collection(scalac_classpath_path_entries_abs)
+          # Step 1: Convert classpath to SemanticDB
+          # ---------------------------------------
+          # If there are any un-metacp'd dependencies, metacp them so their indices can be passed to
+          # Rsc.
```

Comment:
What does un-metacp'd mean?

Reply:
When we are in this context, Java dependencies may not have been run through metacp yet. This section here runs those through metacp. https://github.com/pantsbuild/pants/issues/6754